DEVELOPMENT.md

```diff
@@ -2,88 +2,281 @@
 
 Internal development documentation for OpenHands team.
 
+## Development Setup
+
+### Prerequisites
+- Node.js 18 or higher
+- npm 8 or higher
+- Chrome browser
+- Git
+
+### Initial Setup
+1. Clone the repository
+2. Set up each component:
+
+```bash
+# Set up Pages
+cd pages
+npm install
+npm run build
+
+# Set up Worker
+cd ../worker
+npm install
+npm run build
+
+# Set up Extension
+cd ../extension
+npm install
+npm run build
+```
+
+### Development Workflow
+
+#### Building Components
+
+1. Pages:
+```bash
+cd pages
+npm run build
+```
+
+2. Worker:
+```bash
+cd worker
+npm run build
+```
+
+3. Extension:
+```bash
+cd extension
+npm run build
+```
+
+Note: Running development servers locally is not recommended as they require an interactive terminal and specific environment setup. Instead, use the staging environment for testing.
+
+#### Testing
+
+Each component has its own test suite:
+
+```bash
+# Pages Tests
+cd pages
```

Comment:
I don't like this description at all, let's go back to what we do in the github actions